### PR TITLE
feat: show expired session info

### PR DESCRIPTION
**Pull request by gabaldon**
This issue is a temporal improvement in the user experience when the session expires. The wallet backend should implement a method for extending the session timeout [#1927](https://github.com/witnet/witnet-rust/issues/1927)

### CODE REVIEW

File: src/components/LogoutModal.vue [hunk 1]
```diff
@@ -0,0 +1,93 @@
+<template>
+  <el-dialog
+    class="info"
+    :title="this.$t(close_session_info_title)"
```
Comment (Tommytrg) [inline src/components/LogoutModal.vue:4]:
```suggestion
    :title="$t(close_session_info_title)"
```

File: src/components/LogoutModal.vue [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+    width="30%"
+    :visible="true"
+    :show-close="false"
+    @close="close"
+  >
+    <div slot="title" class="title-container">
+      <font-awesome-icon class="icon" icon="info-circle" />
+      <p class="title">{{ this.$t('close_session_info_title') }}</p>
```
Comment (Tommytrg) [inline src/components/LogoutModal.vue:12]:
```suggestion
      <p class="title">{{ $t('close_session_info_title') }}</p>
```

File: src/components/LogoutModal.vue [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+    </div>
+
+    <p class="text">
+      {{ this.$t('close_session_info_0') }}
+    </p>
+
+    <el-checkbox v-model="checked">{{ this.$t('not_show_again') }}</el-checkbox>
```
Comment (Tommytrg) [inline src/components/LogoutModal.vue:19]:
```suggestion
    <el-checkbox v-model="checked">{{ $t('not_show_again') }}</el-checkbox>
```

File: src/components/LogoutModal.vue [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+    <span slot="footer" class="dialog-footer">
+      <el-button type="primary" @click="close">{{ this.$t('ok') }}</el-button>
```
Comment (Tommytrg) [inline src/components/LogoutModal.vue:21]:
```suggestion
      <el-button type="primary" @click="close">{{ $t('ok') }}</el-button>
```

File: src/store/wallet.js [hunk 5]
```diff
@@ -448,10 +448,18 @@ export default {
     },
     startSessionTimeout(state, ms) {
       // Redirect to wallet list when the session has expired
+      // state.sessionTimeout = setTimeout(() => {
```
Comment (Tommytrg) [inline src/store/wallet.js:451]: Remove comment

File: src/api.js [hunk 6]
```diff
@@ -240,6 +240,14 @@ export class LocalStorageApi {
...
   }
 
+  getSkipLogoutInfo() {
+    return this._get('skipLogoutInfo')
+  }
+
+  setSkipLogoutInfo(val) {
+    return this._set('skipLogoutInfo', val)
+  }
+
```
Comment (Tommytrg) [inline src/api.js:250]: As we are thinking in store more information related to the session expiration I'd rename it with sessionExpirantionPreferences

**Review by aesedepece — CHANGES_REQUESTED** [review]

File: src/store/wallet.js [hunk 7]
```diff
@@ -459,6 +483,23 @@ export default {
     },
   },
   actions: {
+    async refreshSession(context) {
+      const request = await context.state.api.refreshSession({
+        session_id: context.state.sessionId,
+      })
+      console.log(request)
```
Comment (aesedepece) [inline src/store/wallet.js:490]:
```suggestion
```

File: src/store/wallet.js [hunk 8]
```diff
@@ -459,6 +483,23 @@ export default {
...
+      if (request.result.success) {
+        context.state.sessionExtended = true
+        console.log('startSessionTimeout')
```
Comment (aesedepece) [inline src/store/wallet.js:493]:
```suggestion
```

File: src/store/wallet.js [hunk 9]
```diff
@@ -459,6 +483,23 @@ export default {
...
+        context.commit(
+          'startSessionTimeout',
+          context.state.sessionExpirationSecs * 1000,
+        )
+        console.log(request.result)
+      } else {
+        console.log('error while refreshing session')
```
Comment (aesedepece) [inline src/store/wallet.js:500]:
```suggestion
        console.error('error while refreshing session', request.result)
```

**Review by aesedepece — CHANGES_REQUESTED** [review]

**Review by aesedepece — DISMISSED** [review]

**Review by clbartoli — APPROVED** [review]

**Pull request merged**